### PR TITLE
Fixed controller reference deprecations

### LINKED ISSUE #5227: Controllers Should Not Be Referenced With Single Colon Syntax
## Feature Request
Symfony 4.1 deprecates bundle notation (e.g. `SonataAdminBundle:CRUS:export`) - https://symfony.com/blog/new-in-symfony-4-1-deprecated-the-bundle-notation. The change causes a lot of `E_USER_DEPRECATED`:
image

While it doesn't break anything it will be good to fix this by simply replacing `:` with `::` when controllers are references.
This is a similar issue to e.g. https://github.com/FriendsOfSymfony/FOSRestBundle/issues/1897

I'm not that famillar with Sonata codebase, but I see this syntax is really only used in https://github.com/sonata-project/SonataAdminBundle/blob/312593909ca0b21cf723e07660b966a15a385916/src/DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php#L63-L65

### PR DESCRIPTION
**Pull request by DanieleGBX**
I am targeting this branch, because it should be a non-BC fix.

Closes #5227

```markdown
## Changelog
### Fixed
- Fixed controller reference deprecations
```

## Subject
Fixed controller reference deprecations

### CODE REVIEW

File: src/Route/RouteCollection.php [hunk 1]
```diff
@@ -83,7 +83,8 @@ public function add(
         $routeName = $this->baseRouteName.'_'.$name;
 
         if (!isset($defaults['_controller'])) {
-            $defaults['_controller'] = $this->baseControllerName.':'.$this->actionify($code);
+            $actionJoiner = false === \strpos($this->baseControllerName, '\\') ? ':' : '::';
```
Comment (greg0ire) [inline src/Route/RouteCollection.php]: Should we trigger a deprecation here? We are probably not going to support the old notation on next major…

Comment (DanieleGBX) [reply src/Route/RouteCollection.php]: Unluckily I don't know the deprecation policy of the project. But it would be nice to be rid of this silliness.

Comment (greg0ire) [reply src/Route/RouteCollection.php]: @sonata-project/contributors please advise

Comment (greg0ire) [reply src/Route/RouteCollection.php]: IMO we could maybe merge as is and add the deprecation later if really need be

**Comment by greg0ire** [comment]
Triggered the build  again

**Comment by greg0ire** [comment]
It passes :tada: !

File: src/Route/RouteCollection.php (hunk 1 above)
Comment (kunicmarko20) [reply src/Route/RouteCollection.php]: Deprecate for sure, at one point we will just use `::`, right?

File: src/Annotation/Admin.php [hunk 2]
```diff
@@ -78,7 +79,7 @@ class Admin implements MetadataProcessorInterface
     /**
      * @var string
      */
-    public $baseControllerName = 'SonataAdminBundle:CRUD';
+    public $baseControllerName = CRUDController::class;
```
Comment (kunicmarko20) [inline src/Annotation/Admin.php]: Bc break?

**Comment by kunicmarko20** [comment]
@greg0ire the only potential problem I see here is what if someone did:

https://github.com/kunicmarko20/SonataImporterBundle/blob/master/src/DependencyInjection/Compiler/AutoConfigureCompilerPass.php#L67

Basically, If the var is `SonataAdminBundle:CRUD` I mark it as a default Controller. That lib is `0.2` so I don't really care I will change it, but what if someone else did this also?

File: src/Route/RouteCollection.php (hunk 1 above)
Comment (greg0ire) [reply src/Route/RouteCollection.php]: Correct!

Comment (kunicmarko20) [reply src/Route/RouteCollection.php]: hm, true, then skip ti.

Comment (OskarStark) [reply src/Route/RouteCollection.php]: Yes skip it

File: src/Annotation/Admin.php (hunk 2 above)
Comment (greg0ire) [reply src/Annotation/Admin.php]: nah

**Comment by greg0ire** [comment]
@kunicmarko20 cannot think of a way to circumvent that, I'd say merge as is.

**Comment by kunicmarko20** [comment]
Thank you @DanieleGBX !

**Comment by DanieleGBX** [comment]
You're welcome.

File: src/Route/RouteCollection.php (hunk 1 above)
Comment (tifabien) [reply src/Route/RouteCollection.php]: It seems this change causes a BC break. In my admin service configuration I've put as baseControllerName value `AppBundle:Admin\Info` 

Shouldn't we also check for the `:` presence ?

**Pull request merged**